services/apps/src/apps/routes/apps.clj

```diff
@@ -70,6 +70,19 @@
         endpoint to succeed."
         (ok (apps/list-app-permissions current-user (:apps body))))
 
+  (POST* "/sharing" []
+         :query [params SecuredQueryParams]
+         :body [body (describe perms/AppSharingRequest "The app sharing request.")]
+         :return perms/AppSharingResponse
+         :summary "Share one or more apps with one or more users."
+         :description "This endpoint allows the caller to share multiple apps with multiple users. The
+         authenticated user must have ownership permission to every app in the request body for this
+         endpoint to succeed. Note: this is a potentially slow operation and the response is returned
+         synchronously. The DE UI handles this by allowing the user to continue working while the request
+         is being processed. When calling this endpoint, please be sure that the response timeout is long
+         enough."
```

Comment:
What happens if the response timeout isn't long enough for a particular request? As in, what'll get returned, what will and won't happen in the backend, etc.? (I assume the UI should know how to handle that case as well, but that's not this PR)